### PR TITLE
Adding changelog for v0.7.1

### PR DESCRIPTION
**Pull request by robscott**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
This adds the changelog for v0.7.1. By approving this PR, you're approving the release of v0.7.1 based on this commit.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/hold for consensus

### CODE REVIEW

**Review by sunjayBhatia — APPROVED** [review]

**Review by shaneutt — APPROVED** [review]
One question:

/hold

But then lgtm:

/approve
/lgtm

File: CHANGELOG.md [hunk 1]
```diff
@@ -26,6 +27,40 @@
...
+
+- Webhook config works with "restricted" Pod Security level. (#2016, @jcpunk)
+
+### Clarifications
+
+- HTTPRoute Method matching precedence has been clarified. (#2054,
+  @gauravkghildiyal)
+- Implementations MUST ignore any port value specified in the HTTP Host header
+  while performing a match against HTTPRoute.Hostnames. (#1980,
+  @gauravkghildiyal)
```
[the remark this answers is not in the document]
Comment (robscott) [reply CHANGELOG.md:58]: Yep, the same PR actually added a conformance test case to cover this

[the remark this answers is not in the document]
Comment (shaneutt) [reply CHANGELOG.md:58]: Good good, ty

**Comment by shaneutt** [comment]
:ship:

**Pull request merged**